Feature/global search box

imsauravsingh opened the pull request:
-> functional implementation is done
-> Found few issues on typeahead where I am working
-> Test cases are still pending (Pending...)
-> Icon component needs to implement. (it wasn't working as per expectation)

imsauravsingh committed:
Search component created

imsauravsingh committed:
Molecules: Search component build with basic search functionality

imsauravsingh committed:
Molecules: Search component build with basic search functionality

imsauravsingh committed:
voice recognition implemented

imsauravsingh committed:
fixed linting issue

imsauravsingh committed:
Merge branch 'feature/design-system-basics' of https://github.com/pagesource/atomic-react-components into feature/search-box-with-speech-capability

imsauravsingh committed:
Merge branch 'feature/design-system-basics' of https://github.com/pagesource/atomic-react-components into feature/search-box-with-speech-capability

imsauravsingh committed:
speechRecognition hooks created

imsauravsingh committed:
inhanced suggestions

imsauravsingh committed:
change hook folder structure

imsauravsingh committed:
Merge branch 'develop' of https://github.com/pagesource/atomic-react-components into feature/google-search-box

imsauravsingh committed:
implemented flex css and removed duplicate data

imsauravsingh committed:
working on hook test cases

Commit:
resolved conflict

Commit:
Test cases written for hooks

imsauravsingh committed:
Hook test cases implemented

imsauravsingh committed:
Merge branch 'develop' of https://github.com/pagesource/atomic-react-components into feature/google-search-box

imsauravsingh committed:
pr review comment has updated

imsauravsingh committed:
fixed js linting issue

imsauravsingh committed:
Merge branch 'develop' of https://github.com/pagesource/atomic-react-components into feature/google-search-box

imsauravsingh committed:
fixed linting issue

imsauravsingh committed:
resolved conflict

imsauravsingh committed:
Merge branch 'feature/google-search-box' of https://github.com/pagesource/atomic-react-components into feature/global-search-box

imsauravsingh committed:
Merge branch 'develop' of https://github.com/pagesource/atomic-react-components into feature/global-search-box

imsauravsingh committed:
work started on new branch

imsauravsingh committed:
Implementation has been done

vinodloha commented on stylelint.config.js:
```diff
@@ -1,6 +1,6 @@
 const strictRules = {
   'selector-max-specificity': [
-    '0,3,0',
+    '0,4,0',
```
This too much deep selector, we should avoid such scenarios.

vinodloha commented on lib/components/molecules/Search/SearchList/SearchList.js:
```diff
@@ -0,0 +1,33 @@
+import styles from './SearchList.style';
+import List from '../../List';
+
+const SearchList = ({ dataLists, renderList }: Props): Node => {
```
I think this is already available in typeahead

vinodloha commented:
<img width="730" alt="Screenshot 2020-01-07 at 9 01 31 AM" src="https://user-images.githubusercontent.com/8122967/71866660-f790b780-312c-11ea-9be2-0e62f426e54a.png">

UI is broken for search results.

vinodloha reviewed (changes requested):
Reported few issues.

vinodloha commented on lib/components/molecules/Search/Search.js:
```diff
@@ -0,0 +1,110 @@
+          className={classnames('search-txt', inputProps.className)}
+          id="typeahead"
+          name="wordsearch"
+          placeholder="Type any word"
```
@imsauravsingh please don't hard code any text, it should be prop driven

vinodloha commented on lib/components/molecules/Search/Search.js:
```diff
@@ -0,0 +1,110 @@
+      <div className="search-box">
+        <Typeahead
+          className={classnames('search-txt', inputProps.className)}
+          id="typeahead"
```
There should be a way to set the ID and Name from outside. This can be a fallback default value

vinodloha commented on lib/components/molecules/Search/Search.js:
```diff
@@ -0,0 +1,110 @@
+    <div className={classnames(styles.default, className)}>
+      <div className="search-box">
+        <Typeahead
+          className={classnames('search-txt', inputProps.className)}
```
search-txt --> search-text

vinodloha commented on lib/components/molecules/Search/Search.js:
```diff
@@ -0,0 +1,110 @@
+                <i className="search-icon search-loader" />
+              ) : (
+                // eslint-disable-next-line jsx-a11y/control-has-associated-label
+                <i
```
Why we can't use the button component here?

vinodloha commented on lib/components/molecules/Search/Search.js:
```diff
@@ -0,0 +1,110 @@
+          }}
+          renderCustomResultsView={item => <p>{item}</p>}
+          isFieldLabel="true"
+          label="search word"
```
Same here hard coded labels

imsauravsingh committed:
removing conflict from develop branch

Commit:
fix: working on conflict

working on conflict

fix: resolved style conflict

fixed style conflict

fix: pr comment issue

fixed pr comment issue

fix: added flow check in npm validate

fix: added flow as pre push hook

fix: added flow check in pre commit hook

fixed linting issue

fixed linting issue & working on css (flex)

fix: fixed implementation issue on search component

fixed linting and flow errors

fix: search component test cases

fixed test cases issues

imsauravsingh committed:
working to resolved issue on typeahead

imsauravsingh committed:
Fixed design and render props issue

imsauravsingh committed:
fix: fixed snapshot issue

fixed snapshot issue

fix: working on conflict

working on conflict

fix: resolved style conflict

fixed style conflict

fix: pr comment issue

fixed pr comment issue

fix: added flow check in npm validate

fix: added flow as pre push hook

fix: added flow check in pre commit hook

fixed linting issue

fixed linting issue & working on css (flex)

fix: fixed implementation issue on search component

fixed linting and flow errors

fix: search component test cases

fixed test cases issues

fix: pr commit issue

fixed pr commit issue

imsauravsingh committed:
fix: fixed snapshot issue

fixed snapshot
